### ISSUE TITLE
testOnly logic is missing from dist

### ISSUE THREAD

**Issue by jonmellman** [af/envalid#162]
The [`testOnly`](https://github.com/af/envalid#testonly) feature is broken for consumers. When I pass `testOnly`, the behavior is the same as if I passed no `devDefault` at all.

## Repro

The following code throws an error when it shouldn't:
```js
import * as envalid from 'envalid'

envalid.cleanEnv({ NODE_ENV: 'test' }, {
  FOO: envalid.str({
    devDefault: envalid.testOnly('BAR')
  })
})
```

## Investigation

I believe it's because the published `dist/` directory doesn't match the source code.

Investigation steps:
1. Download the latest version from the registry: https://registry.npmjs.org/envalid/-/envalid-7.1.1.tgz
2. Unzip it
3. Compare:
    1. Source code at `package/src/envalid.ts#48` reads:
        ```ts
        export const testOnly = <T>(defaultValueForTests: T) => {
            return process.env.NODE_ENV === 'test' ? defaultValueForTests : ((testOnlySymbol as unknown) as T) // T is not strictly correct, but prevents type errors during usage
        }
        ```
    2. Corresponding dist code at `package/dist/envalid.cjs.development.js#436` reads:
        ```js
        var testOnly = function testOnly(defaultValueForTests) {
            return  testOnlySymbol; // T is not strictly correct, but prevents type errors during usage
        };
        ```

Notice that the `NODE_ENV === 'test'` condition and ternary branch are missing. So, I suspect there's code erasure in your build process. I don't have visibility into the build process to investigate further.

**Comment by af** [comment]
Thanks for the report– I can reproduce and it looks like tsdx (used as of envalid v7) is inlining the result of the ternary. Might try ripping tsdx out to fix this

**Comment by jonmellman** [comment]
Here's a hacky workaround, in case anyone else runs into this:

```ts
const testOnlySymbol = Symbol.for('envalid - test only')
const testOnly = <T>(defaultValueForTests: T) => {
  return process.env.NODE_ENV === 'test' ? defaultValueForTests : ((testOnlySymbol as unknown) as T)
}
```

That version of the `testOnly` utility does work.

**Comment by af** [comment]
Just published `7.2.0-beta.2`, which should resolve this. The TSDX tooling that inlined the code is now completely removed, and `tsc` is now building the published js directly. Please let me know if this works as expected for you

**Comment by jonmellman** [comment]
`7.2.0-beta.2` is working how I expect, thanks for the quick fix!

**Comment by af** [comment]
Fixed and released in 7.2.0
